api/envoy/config/route/v3/route_components.proto

```diff
@@ -1944,3 +1956,15 @@ message InternalRedirectPolicy {
   // x-forwarded-proto. The default is false.
   bool allow_cross_scheme_redirect = 4;
 }
+
+// A simple wrapper for an HTTP filter config. This is intended to be used as a wrapper for the
+// map value in
+// :ref:`VirtualHost.typed_per_filter_config<envoy_api_field_config.route.v3.VirtualHost.typed_per_filter_config>`,
+// :ref:`Route.typed_per_filter_config<envoy_api_field_config.route.v3.Route.typed_per_filter_config>`,
+// or :ref:`WeightedCluster.ClusterWeight.typed_per_filter_config<envoy_api_field_config.route.v3.WeightedCluster.ClusterWeight.typed_per_filter_config>`
+// to indicate that the filter is optional. In this case, if the client does not support the filter,
+// it may ignore the map entry rather than rejecting the config.
+// [#not-implemented-hide:]
+message OptionalFilterConfig {
```

Comment:
A downside of this wrapper is that it adds exactly one feature, and leaves no room for future expansion.  If we need another feature in the future, we'll need two levels of wrapping.  (Actually, we already do have another level of wrapping in the `TypedStruct` feature.)

Instead of this limited-purpose wrapper, maybe we should have one that holds all future expandability instead?

```proto
message FilterConfig {
  oneof config_type {
    proto.Any typed_config = 1;  // Would never contain a TypedStruct
    udpa.type.v1.TypedStruct struct_config = 2;
    config.core.v3.ExtensionConfigSource config_discovery = 3; // ? (from HttpFilter)
  }
  bool optional = 4;
  // room for future expansion
}
```

Also, optionally & more invasively, we could update `HttpFilter` to deprecate `config_type` and reference this new `FilterConfig` message to unify things further.

Reply:
I like the idea of making this more general-purpose so that we can add additional per-filter options later!  Done.

I didn't include the `TypedStruct` field, because I think the established way of dealing with `TypedStruct` in xDS is to embed it inside of an `Any` field, since that makes the JSON representation more natural.

I also didn't include the `ExtensionConfigSource` field, since I don't see any existing option for using ECDS here in the `typed_per_filter_config` fields.  I suspect the two are mutually exclusive, but @htuch can confirm.